packages/eslint-plugin/src/rules/await-thenable.ts

```diff
@@ -79,11 +79,18 @@ export default createRule<[], MessageId>({
           return;
         }
 
-        const asyncIteratorSymbol = tsutils.getWellKnownSymbolPropertyOfType(
-          type,
-          'asyncIterator',
-          checker,
-        );
+        const types = type.isUnion() ? type.types : [type];
```

Comment:
(nit) we use `tsutils.unionTypeParts()` for this

https://joshuakgoldberg.github.io/ts-api-utils/stable/functions/unionTypeParts.html

Reply:
Good to know, thanks !

Reply:
done